### PR TITLE
Split browser tests into categories

### PR DESCRIPTION
**Pull request by NejcZdovc**
Resolves https://github.com/brave/brave-browser/issues/9204

## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [ ] Android
  - [ ] iOS
  - [ ] Linux
  - [x] macOS
  - [ ] Windows
- Verified that these changes pass automated tests (unit, browser, security tests) on
  - [ ] iOS
  - [ ] Linux
  - [x] macOS
  - [ ] Windows
- [ ] Verified that all lint errors/warnings are resolved (`npm run lint`)
- [ ] Ran `git rebase master` (if needed).
- [ ] Ran `git rebase -i` to squash commits (if needed).
- [ ] Tagged reviewers and labelled the pull request as needed.
- [ ] Request a security/privacy review as needed.
- [ ] Add appropriate QA labels (QA/Yes or QA/No) to include the closed issue in milestone
- [ ] Public documentation has been updated as necessary. For instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-compatibility-issues-with-tracking-protection
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions 
- [ ] Verify test plan is specified in PR before merging to source

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on.
- [ ] All relevant documentation has been updated.

### CODE REVIEW

File: components/brave_rewards/browser/test/common/rewards_browsertest_network_util.cc [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+        "status": "%s",
+        "currencies": ["BAT"]
+      })",
+      name.c_str(),
+      verified.c_str(),
+      status.c_str());
+}
+
+std::string GetUpholdCard(
+    const std::string balance,
```
Comment (emerick) [inline components/brave_rewards/browser/test/common/rewards_browsertest_network_util.cc:124]: Should have an `&` here.

File: components/brave_rewards/browser/test/common/rewards_browsertest_contribution.cc [hunk 2]
```diff
@@ -0,0 +1,469 @@
...
+
+  rewards_service_->AddObserver(this);
+}
+
+content::WebContents* RewardsBrowserTestContribution::contents() {
+  return browser_->tab_strip_model()->GetActiveWebContents();
+}
+
+void RewardsBrowserTestContribution::TipViaCode(
+    const std::string publisher_key,
```
[the remark this answers is not in the document]
Comment (NejcZdovc) [reply components/brave_rewards/browser/test/common/rewards_browsertest_contribution.cc:36]: fixed

File: components/brave_rewards/browser/test/common/rewards_browsertest_contribution.h [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+  ledger::Result tip_reconcile_status_ = ledger::Result::LEDGER_ERROR;
+  bool pending_tip_saved_ = false;
+  std::unique_ptr<base::RunLoop> wait_for_pending_tip_saved_loop_;
+  bool recurring_tip_saved_ = false;
+  std::unique_ptr<base::RunLoop> wait_for_recurring_tip_saved_loop_;
+  bool multiple_tip_reconcile_completed_ = false;
+  std::unique_ptr<base::RunLoop> wait_for_multiple_tip_completed_loop_;
+  int32_t multiple_tip_reconcile_count_ = 0;
+  int32_t multiple_tip_reconcile_needed_ = 0;
+  std::vector<ledger::Result> multiple_tip_reconcile_status_;
```
[the remark this answers is not in the document]
Comment (NejcZdovc) [reply components/brave_rewards/browser/test/common/rewards_browsertest_contribution.h:135]: fixed

File: components/brave_rewards/browser/test/common/rewards_browsertest_contribution.h [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+  bool multiple_ac_reconcile_completed_ = false;
+  std::unique_ptr<base::RunLoop> wait_for_multiple_ac_completed_loop_;
+  int32_t multiple_ac_reconcile_count_ = 0;
+  int32_t multiple_ac_reconcile_needed_ = 0;
+  std::vector<ledger::Result> multiple_ac_reconcile_status_;
```
[the remark this answers is not in the document]
Comment (NejcZdovc) [reply components/brave_rewards/browser/test/common/rewards_browsertest_contribution.h:140]: fixed

File: components/brave_rewards/browser/test/common/rewards_browsertest_network_util.cc [hunk 5]
```diff
@@ -0,0 +1,184 @@
...
+#include "brave/components/brave_rewards/browser/test/common/rewards_browsertest_network_util.h"
+#include "content/public/test/browser_test_utils.h"
+
+namespace rewards_browsertest_util {
+
+bool URLMatches(
+    const std::string& url,
+    const std::string& path,
+    const std::string& prefix,
+    const braveledger_request_util::ServerTypes& server) {
```
[the remark this answers is not in the document]
Comment (NejcZdovc) [reply components/brave_rewards/browser/test/common/rewards_browsertest_network_util.cc:19]: fixed

File: components/brave_rewards/browser/test/common/rewards_browsertest_contribution.h [hunk 6]
```diff
@@ -0,0 +1,150 @@
...
+      const bool should_contribute = false,
+      const bool recurring = false,
+      const ledger::Result result = ledger::Result::LEDGER_OK);
+
+  void TipPublisher(
+      const GURL& url,
+      rewards_browsertest_util::ContributionType type,
+      bool should_contribute = false,
+      int32_t selection = 0,
+      int32_t number_of_contributions = 1);
```
[the remark this answers is not in the document]
Comment (NejcZdovc) [reply components/brave_rewards/browser/test/common/rewards_browsertest_contribution.h:45]: great suggestion, fixing

**Pull request merged**